### DISCUSSION TITLE
Windows Defender flagging my IronScheme REPL app, sometimes, but always when I'm defining a variable....

### DISCUSSION THREAD

**Discussion by bcalco** [leppie/IronScheme#119] [Discussions]
So, lately I've been having issues with Windows Defender on my dev machine. 

At least two or three times recently I and my testers have run into a situation where, in the middle of working in the IronScheme REPL (started by my wrapper app), Windows Defender would pop up with what must be a false positive. Among other reasons I am sure it's erroneous, it picks a different Trojan name as matching each time. Also, it's *always* when using `(define ...)`, so I guess what's happening is behind the scenes some fugly identifier name is being created by IS, and the AV runtime snoop is picking up some blunt regex match (most recently, a cryptomining trojan LOL) and throwing the false alarm.

This has been happening with increasing frequency to my .NET artifacts. Mind you, it doesn't find anything in the assemblies I'm using (with one exceptions which I've stopped using, and the author of that says the only answer is to keep submitting the false positive to MS till they figure out why they're doing it). Nor does it complain after I compile. I just happens "at random" when I define a term, and never the same term trips it (I tend to use a handful of common identifiers when REPLing. Sometimes they trigger it, sometimes they don't.

a) has anyone besides me been seeing this?

b) @leppie - have you any insight into what might be happening, and how to make it stop?

We went out and purchased an expensive EV cert to get past "SmartScreen" (lol) and the apps we work with that require certs, but AV seems to be not liking my IronScheme app, lately.

**Comment by bcalco** [comment]
Today it keeps finding some flavor of `Script/Sabsik.TE.A!ml` each time I enter a REPL (another is `Script/Sabsik.FL.A!ml`). Very annoying. Funny thing is, after I let it delete my app, it can't find it on my computer at all. Once I compile it, it finds it. Only when I'm in the REPL. Grrr.

**Comment by bcalco** [comment]
So, now it's not showing any issues after a couple scans and I am not sure what to make of it. 

I posted this here in case anyone else perchance had a similar issue (I did have a similar issue with another framework I was using), but you can delete it, @leppie. I would do it but I don't see how, prolly cuz I don't have permissions.

**Comment by leppie** [comment]
I have never come across this. 

Really interested what triggers it though.

**Reply by bcalco** [reply]
So am I! 

My anecdotal observation is that once Defender has a chance to scan the build, it's not as easily spooked by whatever is triggering its pattern matching algorithms at runtime. So now I'm doing "quick scans" after (almost) every build before I try to run it, which feels absurd, but so far I've not run into it again for a day and a half. I almost feel productive again. It's like having to get a PCR test every time I walk in a pub (which used to be frequently, but now is almost never, thanks to various neurotic policies of the sort that make it no longer fun), and I hate it, but at least it's not taking my EXE away anymore.
